### PR TITLE
isort: Support disabling sections with ``no-sections = true``

### LINKED ISSUE #8653: support no_sections option for isort
ruff's isort doesn't support the ``no_sections`` option that isort upstream has.  This option puts all import sinto the same import bucket.

For example, consider this source file:

```python
import os

import bar
import blah
```

isort --no-sections x.py format it as

```python
import bar
import blah
import os
```

As far as I can tell, there is no set of options for ruff's isort that provides this behaviour (if there is and I've missed it, I'll volunteer to add a note to the docs)

### PR DESCRIPTION
**Pull request by jelmer**
## Summary

This adds a ``no-sections`` option for isort in the linter, similar to the ``no_sections`` option that exists in upstream isort (https://pycqa.github.io/isort/docs/configuration/options.html#no-sections)

This option puts all imports except for ``__future__`` into the same section, and is mostly used by monorepos.

I've taken a bit of a leap in assuming that ruff wants to support the exact same option; more than happy to refactor if you'd prefer a different way of setting this up.

Fixes #8653

## Test Plan

I've added a test and have run it on a large Python codebase that uses isort with --no-sections. The option is disabled by default.

### CODE REVIEW

File: crates/ruff_linter/src/rules/isort/categorize.rs [hunk 1]
```diff
@@ -72,16 +73,22 @@ pub(crate) fn categorize<'a>(
...
     let module_base = module_name.split('.').next().unwrap();
     let (import_type, reason) = {
-        if level.is_some_and(|level| level > 0) {
+        if !no_sections && level.is_some_and(|level| level > 0) {
             (
                 &ImportSection::Known(ImportType::LocalFolder),
                 Reason::NonZeroLevel,
             )
         } else if module_base == "__future__" {
             (&ImportSection::Known(ImportType::Future), Reason::Future)
```
Comment (charliermarsh) [inline crates/ruff_linter/src/rules/isort/categorize.rs:86]: Does isort put the `__future__` imports in their own section, even if you pass `--no-sections`?

File: crates/ruff_workspace/src/options.rs [hunk 2]
```diff
@@ -2107,6 +2117,9 @@ impl IsortOptions {
             }
         }
 
+        let no_sections = self.no_sections.unwrap_or_default();
+        // TODO: Warn if `no_sections` is true and `sections` or `section-order` is non-empty?
```
Comment (charliermarsh) [inline crates/ruff_workspace/src/options.rs:2121]: I think it'd be great to have some of these warnings here prior to merging.

**Review by charliermarsh — COMMENTED** [review]
Thanks for getting involved! No objection to adding this, and no objection to using the same nomenclature and API as isort. Left a few small questions and comments, but should be able to get this merged relatively quickly.

File: crates/ruff_workspace/src/options.rs (hunk 2 above)
Comment (jelmer) [reply crates/ruff_workspace/src/options.rs]: I've added a warning for these two situations.

**Review by charliermarsh — APPROVED** [review]
Thanks!

**Pull request merged**